<a href="https://colab.research.google.com/github/andremarinho17/data_analytics_projects_en/blob/main/Activity_Explore_descriptive_statistics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Activity: Explore descriptive statistics - André Marinho

## Applying descriptive statistics for the EPA to investigate the air quality

**Author**: André Marinho

<p align="center"><img src="https://weadapt.org/wp-content/uploads/2023/05/screenshot_2022-10-19_at_16.35.53.png" height="150px">



## **Introduction**

Data professionals often use descriptive statistics to understand the data they are working with and provide collaborators with a summary of the relative location of values in the data, as well an information about its spread.

For this activity, you are a member of an analytics team for the United States Environmental Protection Agency (EPA). You are assigned to analyze data on air quality with respect to carbon monoxide, a major air pollutant. The data includes information from more than 200 sites, identified by state, county, city, and local site names. You will use Python functions to gather statistics about air quality, then share insights with stakeholders.

## **Step 1: Imports**


Import the relevant Python libraries `pandas` and `numpy`.

In [ ]:
# Import relevant Python libraries.

import pandas as pd
import numpy as np

The dataset provided is in the form of a .csv file named `c4_epa_air_quality.csv`. It contains a subset of data from the U.S. EPA. As shown in this cell, the dataset has been automatically loaded in for you. You do not need to download the .csv file, or provide more code, in order to access the dataset and proceed with this lab. Please continue with this activity by completing the following instructions.

In [ ]:
# RUN THIS CELL TO IMPORT YOUR DATA.

df = pd.read_csv("c4_epa_air_quality.csv", index_col = 0)

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about loading data in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `pandas` library that allows you to read in data from a .csv file and load it into a DataFrame.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `read_csv` function from the pandas `library`. The `index_col` parameter can be set to `0` to read in the first column as an index (and to avoid `"Unnamed: 0"` appearing as a column in the resulting DataFrame).

</details>

## **Step 2: Data exploration**

To understand how the dataset is structured, display the first 10 rows of the data.

In [ ]:
# Display first 10 rows of the data.

df.head()

date_local    state_name   county_name      city_name  \
0  2018-01-01       Arizona      Maricopa        Buckeye   
1  2018-01-01          Ohio       Belmont      Shadyside   
2  2018-01-01       Wyoming         Teton  Not in a city   
3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158    5  
2  Parts per million         0.111111    2  
3  Parts per million         0.300000    3  
4  Parts per million         0.215789    3

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about exploratory data analysis in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `pandas` library that allows you to get a specific number of rows from the top of a DataFrame.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `head()` function from the `pandas` library.

</details>

**Question:** What does the `aqi` column represent?

It does represent the level of air quality.

Now, get a table that contains some descriptive statistics about the data.

In [ ]:
df.describe()

arithmetic_mean         aqi
count       260.000000  260.000000
mean          0.403169    6.757692
std           0.317902    7.061707
min           0.000000    0.000000
25%           0.200000    2.000000
50%           0.276315    5.000000
75%           0.516009    9.000000
max           1.921053   50.000000

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `pandas` library that allows you to generate a table of basic descriptive statistics about the numeric columns in a DataFrame.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `describe()` function from the `pandas` library.

</details>

**Question:** Based on the table of descriptive statistics, what do you notice about the count value for the `aqi` column?

The aqi column contains 260 entries and has an average of 6.7.

**Question:** What do you notice about the 25th percentile for the `aqi` column?

This is an important measure for understanding where the aqi values lie.

The 25th percentile value is 2, which means that 25% of the values in this dataset are below 2.

**Question:** What do you notice about the 75th percentile for the `aqi` column?

This is another important measure for understanding where the aqi values lie.

The value of the 25th percentile is 9, which means that 25% of the values of this dataset are below 9.

## **Step 3: Statistical tests**

Next, get some descriptive statistics about the states in the data.

In [ ]:
# Get descriptive stats about the states in the data.

df['state_name'].describe()

count            260
unique            52
top       California
freq              66
Name: state_name, dtype: object

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `pandas` library that allows you to generate basic descriptive statistics about a DataFrame or a column you are interested in.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

 Use the `describe()` function from the `pandas` library. Note that this function can be used:
- "on a DataFrame (to find descriptive statistics about the numeric columns)"
- "directly on a column containing categorical data (to find pertinent descriptive statistics)"

</details>

**Question:** What do you notice while reviewing the descriptive statistics about the states in the data?

Note: Sometimes you have to individually calculate statistics. To review to that approach, use the `numpy` library to calculate each of the main statistics in the preceding table for the `aqi` column.

There are 52 unique states and California is the most frequent state in the data, with a frequency of 66.

## **Step 4. Results and evaluation**

Now, compute the mean value from the `aqi` column.

In [ ]:
# Compute the mean value from the aqi column.

df['aqi'].mean()

6.757692307692308

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the mean value from an array or a Series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `mean()` function from the `numpy` library.

</details>

**Question:** What do you notice about the mean value from the `aqi` column?

This is an important measure, as it tells you what the average air quality is based on the data.

The average value of the air quality is 6.

Next, compute the median value from the aqi column.

In [ ]:
# Compute the median value from the aqi column.

df['aqi'].median()

5.0

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the median value from an array or a series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `median()` function from the `numpy` library.

</details>

**Question:** What do you notice about the median value from the `aqi` column?

This is an important measure for understanding the central location of the data.

The middle value of the air quality is 5.

Next, identify the minimum value from the `aqi` column.

In [ ]:
# Identify the minimum value from the aqi column.

df['aqi'].min()

0

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the minimum value from an array or a Series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `min()` function from the `numpy` library.

</details>

**Question:** What do you notice about the minimum value from the `aqi` column?

This is an important measure, as it tell you the best air quality observed in the data.

The minimum value of the air quality is 0.

Now, identify the maximum value from the `aqi` column.

In [ ]:
# Identify the maximum value from the aqi column.

df['aqi'].max()

50

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the maximum value from an array or a Series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `max()` function from the `numpy` library.

</details>

**Question:** What do you notice about the maximum value from the `aqi` column?

This is an important measure, as it tells you which value in the data corresponds to the worst air quality observed in the data.

The maximum value of the air quality is 50.

Now, compute the standard deviation for the `aqi` column.

By default, the `numpy` library uses 0 as the Delta Degrees of Freedom, while `pandas` library uses 1. To get the same value for standard deviation using either library, specify the `ddof` parameter to 1 when calculating standard deviation.

In [ ]:
# Compute the standard deviation for the aqi column.

df['aqi'].std()

7.0617066788207215

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video section about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the standard deviation from an array or a series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

Use the `std()` function from the `numpy` library. Make sure to specify the `ddof` parameter as 1. To read more about this function,  refer to its documentation in the references section of this lab.

</details>

**Question:** What do you notice about the standard deviation for the `aqi` column?

This is an important measure of how spread out the aqi values are.

The standard deviation of the air quality is 7.06. Given that the average is 6.75, it's possible to infer that the values of the dataset are close to the average value.

## **Considerations**


**What are some key takeaways that you learned during this lab?**

Pandas and numpy libraries are very useful for calculating descriptive statistics in Python and generating insights. Functions such as describe, max, min, std, and median can be significantly important through this process.

**How would you present your findings from this lab to others? Consider the following relevant points noted by AirNow.gov as you respond:**
- AQI values at or below 100 are generally thought of as satisfactory. When AQI values are above 100, air quality is considered unhealthy—at first for certain sensitive groups of people, then for everyone as AQI values increase.
- An AQI of 100 for carbon monoxide corresponds to a level of 9.4 parts per million.

The air quality in most states is safe and healthy, given that the AQI average is 6 and that 75% of the values are below 9. Furthermore, the largest value is 50, which is way below 100.

**What summary would you provide to stakeholders? Use the same information provided previously from AirNow.gov as you respond.**

* The air quality in most states is safe and healthy, given that the AQI average is 6 and that 75% of the values are below 9.
* An investigation into the less unhealthy states should be conducted in order to improve the air quality.

**References**

[Air Quality Index - A Guide to Air Quality and Your Health](https://www.airnow.gov/sites/default/files/2018-04/aqi_brochure_02_14_0.pdf). (2014,February)

[Numpy.Std — NumPy v1.23 Manual](https://numpy.org/doc/stable/reference/generated/numpy.std.html)

US EPA, OAR. (2014, 8 July).[*Air Data: Air Quality Data Collected at Outdoor Monitors Across the US*](https://www.epa.gov/outdoor-air-quality-data).

**Congratulations!** You've completed this lab. However, you may not notice a green check mark next to this item on Coursera's platform. Please continue your progress regardless of the check mark. Just click on the "save" icon at the top of this notebook to ensure your work has been logged.